start.py

```diff
@@ -1,40 +1,100 @@
-from board import Board, EndStates, SpotStates, BoardState, WinningCombos 
+from board import Board, EndStates, SpotStates, BoardState, WinningCombos
 from user import User, UserActions
+from ai import Ai
 import board
 from ui import Ui, BoardPresenter, CommandLinePrompt
 
-def start_game(current_state, player1, player2):
-    Ui.msg(BoardPresenter.display_terminal_board(current_state))
-    
-    if EndStates.is_draw(current_state):
-        Ui.msg('DRAW. GameOver')
-        return True
-     
-    response = CommandLinePrompt.get_input("Enter a number from 1-9: ") 
-    response = UserActions.make_move(current_state, response)
-     
-    if response == False:
-        start_game(current_state, player1, player2)
-
-    # update the board state
-    BoardState.update_state(current_state, User.current_player, response)
-     
-    if EndStates.did_a_player_win(current_state, User.current_player, win_config.winning_combos):
-        Ui.msg(BoardPresenter.display_terminal_board(current_state)) 
-        Ui.msg('Game Over: ' + User.current_player + ' WINS!')
-        return True
-
-    if User.current_player == player1:
-        User.switch_current_user(User.current_player, player2)
-    else:
-        User.switch_current_user(User.current_player, player1)
-    start_game(current_state, player1, player2)
+class GameModes:
```

Comment:
If you write the same line of code more than once, you should go 'hmm, i might need to refactor this later to reduce duplication'. If you write it three or more times, it's time to refactor. There are large blocks of duplicate code here.